File: leetcode/Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        bool neg = x < 0 ? true : false;
        x = neg ? -x : x;
        int digits[30];
        int base = 10, i = 0;
        while (x) {
            digits[i++] = x % base;
            x /= base;
        }
        int res = 0;
        for (int j = 0; j < i; j++) {
            res += digits[j];
            if (j != i-1)
                res *= base;
        }
        return neg ? -res : res;
    }
};

File: leetcode/Reverse Integer.java
public class Solution {
    public int reverse(int x) {
		int X = Math.abs(x);
		int ans = 0;
		while (X > 0){
			ans = ans * 10 + X % 10;
			X = X / 10;
		}
		return (x > 0) ? ans : -ans;
    }
}